hazelcast/src/test/java/com/hazelcast/query/SqlPredicateTest.java

```diff
@@ -67,13 +67,100 @@
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertSame;
 import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
 
 @RunWith(HazelcastSerialClassRunner.class)
 @Category(QuickTest.class)
 public class SqlPredicateTest {
 
     final InternalSerializationService serializationService = new DefaultSerializationServiceBuilder().build();
 
+    static final String[] TEST_MATCHING_SQL_PREDICATES = new String[] {
+            "name = 'Joe' and age = 25 and (city = 'austin' or city = 'AUSTIN')",
+            "name = 'Joe' or city = 'Athens'",
+            "(name = 'Jane' or name = 'Joe' or city = 'AUSTIN') and age = 25",
+            "(name = 'Jane' or name = 'Joe' or city = 'AUSTIN') and age = 25 and salary = 0",
+            "(name = 'Jane' or name = 'Joe') and age = 25 and salary = 0 or age = 24",
+            "name = 'Jane' or age = 25 and name = 'Joe'", // correct precedence is "name = 'Jane' or (age = 25 and name = 'Joe')
+            "age = 35 or age = 24 or age = 31 or (name = 'Joe' and age = 25)",
+            };
+
+    static final String[] TEST_NOT_MATCHING_SQL_PREDICATES = new String[] {
+            "name = 'Joe' and age = 21 and (city = 'austin' or city = 'ATHENS')",
+            "name = 'Jane' or city = 'Athens'",
+            "(name = 'Jane' or name = 'Catie' or city = 'San Jose') and age = 25",
+            "(name = 'Joe' or name = 'Catie' or city = 'San Jose') and age = 21",
+            "(name = 'Jane' or name = 'Joe' or city = 'AUSTIN') and age = 25 and salary = 10",
+            "(name = 'Jane' or name = 'Catie' or city = 'San Jose') and age = 25 and salary = 10",
+            "(name = 'Jane' or name = 'Joe') and age = 25 and salary = 13 or age = 24",
+            "name = 'Jane' or age = 25 and name = 'Catie'",
+            "age = 35 or age = 24 or age = 31 or (name = 'Joe' and age = 27)",
+            };
+
+    @Test
+    public void testSqlPredicates() {
+        Employee employee = new Employee("Joe", "AUSTIN", 25, true, 0);
+        for (String s : TEST_MATCHING_SQL_PREDICATES) {
+            try {
+                assertSqlTrue(s, employee);
+            }
+            catch (AssertionError assertionError) {
```

Comment:
I'd do following:
- rename `assertSqlTrue()` to `assertSqlMatching()`.
- amend `assertSqlMatching()` to include the SQL on failure: `assertTrue("Failed matching SQL predicate \"" + s + '"', predicate.apply(entry));`
- remote the try/catch block from here - no need to polute a test with it
- do the same for `assertSqlFalse()`
WDYT?